tibco_ems/tests/test_unit.py

```diff
@@ -0,0 +1,158 @@
+from datadog_checks.dev.utils import get_metadata_metrics
+from datadog_checks.tibco_ems import TibcoEMSCheck
+
+from .common import METRIC_DATA, SECTION_OUTPUT_SHOW_ALL, SECTION_RESULT, SHOW_MAP, mock_output
+
+
+# Test the main check functionality
+def test_check(dd_run_check, aggregator, instance):
+    check = TibcoEMSCheck('tibco_ems', {}, [instance])
+    check.run_tibco_command = MagicMock(return_value=mock_output('show_all'))
+    dd_run_check(check)
+
+    for metric in METRIC_DATA:
+        aggregator.assert_metric(metric)
+        aggregator.assert_metric_has_tag(metric, 'server_version:10.1.0')
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+    aggregator.assert_all_metrics_covered()
+
+
+# Test specific show server metrics
+def test_show_server_metrics(dd_run_check, aggregator, instance):
+    expected_metrics = SHOW_MAP['show server']['expected_metrics']
+
+    check = TibcoEMSCheck('tibco_ems', {}, [instance])
+    check.run_tibco_command = MagicMock(return_value=mock_output('show_server'))
+    dd_run_check(check)
+
+    for metric in expected_metrics:
+        aggregator.assert_metric(metric)
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+
+# Test the parse_show_server method
+def test_parse_show_server():
+    section = SHOW_MAP['show server']['section']
+    regex = SHOW_MAP['show server']['regex']
+
+    check = TibcoEMSCheck('tibco_ems', {}, [{}])
+    output = check._section_output(section)['show server']
+    result = check._parse_show_server(output, regex)
+
+    expected_result = SHOW_MAP['show server']['expected_result']
+
+    assert result == expected_result
+
+
+# Parametrized test for different show commands
+@pytest.mark.parametrize(
+    "output, expected_metrics",
+    [
+        pytest.param(
+            mock_output('show_queues'),
+            SHOW_MAP['show queues']['expected_metrics'],
+            id="show queues",
+        ),
+        pytest.param(
+            mock_output('show_topics'),
+            SHOW_MAP['show topics']['expected_metrics'],
+            id="show topics",
+        ),
+        pytest.param(
+            mock_output('show_stat_consumers'),
+            SHOW_MAP['show stat consumers']['expected_metrics'],
+            id="show stat consumers",
+        ),
+        pytest.param(
+            mock_output('show_stat_producers'),
+            SHOW_MAP['show stat producers']['expected_metrics'],
+            id="show stat producers",
+        ),
+        pytest.param(
+            mock_output('show_connections'),
+            SHOW_MAP['show connections full']['expected_metrics'],
+            id="show connections full",
+        ),
+        pytest.param(
+            mock_output('show_durables'),
+            SHOW_MAP['show durables']['expected_metrics'],
+            id="show durables",
+        ),
+    ],
+)
+def test_show_metrics(dd_run_check, aggregator, instance, output, expected_metrics):
```

Comment:
💪 for adding this!! It looks like it covers the same ground as `test_parse_factory` in terms of tibco output, is that correct?

Reply:
I think validating the structure of the returned object is important in `test_parse_factory` since we're not validating a metric value but rather just the presence of the metric. I think keeping it gives us immediate feedback into any breaking changes we might make in the parsing that may not be all that apparent in `test_show_metrics`.